src/DataRenderer.tsx

```diff
@@ -59,7 +62,9 @@ function ExpandableObject({
     if (!shouldExpandNodeCalledRef.current) {
       shouldExpandNodeCalledRef.current = true;
     } else {
-      setExpanded(shouldExpandNode(level, value, field));
+      const shouldExpand = shouldExpandNode(level, value, field);
+      onNodeExpandToggled?.(shouldExpand, level, value, field);
```

Comment:
Thank you! I think code is almost ready to be merged.
One part I still worry about is if we change the `shouldExpandNode` property but no node actually changes the state it will anyway cause `onNodeExpandToggled` calls for every node.
I think it would be better to check if current `expand` value is different from the new `shouldExpand`.